lib/src/global_edf_scheduler.rs

```diff
@@ -0,0 +1,211 @@
+use petgraph::graph::Graph;
+
+use std::cmp::Ordering;
+
+use crate::getset_dag_set_scheduler;
+use crate::graph_extension::GraphExtension;
+use crate::scheduler::NodeDataWrapper;
+use crate::{
+    graph_extension::NodeData, homogeneous::HomogeneousProcessor, log::DAGSetSchedulerLog,
+    processor::ProcessorBase, scheduler::DAGSetSchedulerBase,
+};
+
+impl PartialOrd for NodeDataWrapper {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        // Define the keys to compare
+        let key1 = "absolute_deadline";
+        let key2 = "dag_id";
+
+        let self_val = self.node_data.params.get(key1).unwrap();
+        let other_val = other.node_data.params.get(key1).unwrap();
+
+        match self_val.cmp(other_val) {
+            // If the keys are equal, compare by id
+            Ordering::Equal => match self.node_data.id.partial_cmp(&other.node_data.id) {
+                // If the ids are also equal, compare by dag_id
+                Some(Ordering::Equal) => {
+                    let self_dag = self.node_data.params.get(key2).unwrap();
+                    let other_dag = other.node_data.params.get(key2).unwrap();
+                    Some(self_dag.cmp(other_dag))
+                }
+                other => other,
+            },
+            other => Some(other),
+        }
+    }
+}
+
+impl Ord for NodeDataWrapper {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.partial_cmp(other).unwrap_or(Ordering::Equal)
+    }
+}
```

Comment:
This process should be moved to scheduler.rs.

Reply:
Moved.